### PR TITLE
Document metric classes

### PR DESCRIPTION
**Pull request by jerryct**

### CODE REVIEW

**Comment by jerryct** [comment]
Or clang-doc (https://clang.llvm.org/extra/clang-doc.html) - at one point in time.

**Comment by gjasny** [comment]
please rebase to fix TravisCI

**Comment by jerryct** [comment]
I rebased it.

File: core/include/prometheus/histogram.h [hunk 1]
```diff
@@ -6,21 +6,52 @@
...
+  /// The bucket boundaries cannot be changed once the histogram is created.
+  Histogram(const BucketBoundaries&);
+
+  /// \brief Observe the given amount.
+  ///
+  /// The given amount selects the 'observed' bucket. The observed bucket is
+  /// chosen for which the given amount falls into the half-open interval [b_n,
+  /// b_n+1). The counter of the observed bucket is incremented. Also the total
+  /// sum of all observations is incremented.
+  void Observe(double);
```
Comment (gjasny) [inline core/include/prometheus/histogram.h]: Is there a reason behind removing the argument names?

Comment (jerryct) [reply core/include/prometheus/histogram.h]: It is the style currently used in many places of the library in the header file. Personally I would add the argument names also in the header (it also eases documenting with doxygen). I reverted the changes made to the argument names.

**Comment by gjasny** [comment]
Do you have an idea where we could publish the generated doxygen files easily? I played around with "Read The Docs" [here](https://prometheus-cpp.readthedocs.io/en/read-the-docs/) and Breathe / Sphinx, but it does not look promising. Uploading to github pages might also be an option, but maybe you know something easier to set up?

**Comment by gjasny** [comment]
Could you please rebase?

**Comment by jerryct** [comment]
I rebased it.

**Comment by jerryct** [comment]
Regarding documentation I have no better idea. I think github pages is the most straightforward way. Travis simply generates the html and uploads it to a special branch. There are some examples:

https://gist.github.com/francesco-romano/351a6ae457860c14ee7e907f2b0fc1a5

Why don't you like the read the docs output? We need to pimp it a little bit further:
- don't generate private/protected members
- exclude namespace detail with a doxygen annotation
...

So I would still give it a try.

**Comment by jerryct** [comment]
As a starting point I added a Doxygen configuration:

https://github.com/jupp0r/prometheus-cpp/pull/189

You can run it with `doxygen doxygen.config` in the project source dir. I disabled many things in order to get only the public API. For example everything in namespace detail or directory detail is excluded. Can you try it with read the docs?

**Comment by jerryct** [comment]
I think the easiest way to get the documentation is GitHub pages. We could simply create a new repository and let travis push to it:

https://github.com/jerryct/doc_test
https://jerryct.github.io/doc_test/index.html

**Pull request merged**